OAK-8328 | Config errors in index defintion causes index update to ge…

nit0906 opened the pull request:
…t stuck in a loop

stillalex committed:
OAK-8142 CompositePrincipalProvider support for full text search



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856464 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8176: Inaccurate calculation of RevisionVector memory

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856465 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-301: Document Oak
Memoirs in Garbage Collection:
- Improved cross referencing into GitHub
- Paragraph about how reachability is used for bulk segments in Oak 1.6 to determine reclaimability for those

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856466 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8116 Expose text extraction metrics as sling metrics

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856472 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7731 Order by jcr:score descending is not always ignored

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856481 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7731 Order by jcr:score descending is not always ignored

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856483 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8179: Update jacoco to 0.8.3

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856538 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8180: Update mockito to 2.25.1

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856545 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8036 : Improve test in oak-security-spi

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856551 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
minor improvement: proper usage of ConfigurationParams in oak-exercise, defaultvalue may only be null if class type is specified

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856795 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8036 : Improve test in oak-security-spi (2 tests failing with java 1.8)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856814 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
OAK-8133 - Word SHA1 no longer allowed

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856818 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8188 : Adobe internal jenkins finds lower branch coverage

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856828 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8178 : Document missing support for multiplexing in Cug Authorization

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856831 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8134 Index Copier Stats MBean shows stale info

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856854 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8189 Getting warning message in error logs while trying to search using filters

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856862 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8177 Add logging around getReferenceCheckpoint method

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856879 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8138: In cases where search hit count is less than sample size defined for statistical mode for facets, secure mode should be used so that the counts are accurate for non-admin users.

Applying patch contributed by Nitin Gupta

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856883 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8055 Add conflict handler for rep:lastSynced property on external groups



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856908 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-301: Document Oak
Memoirs in Garbage Collection (WIP):
- Describe checkpoint rebasing and tail compaction in Oak 1.8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856909 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
OAK-8193 - Update docs with new release strategies

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856913 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
OAK-6880 - Build fails on a fresh m2 repo

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856930 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8182 : Improvements to PasswordUtil

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856938 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8045 : Allow for ranking being specified with UserAuthenticatonFactory implementations

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856940 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8195 : AutoSaveEnabledManager.createSystemUser calls createUser on delegatee

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856959 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-6880: change 1856930 undone

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1856961 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8196: Update httpclient/mime dependencies to 4.5.8

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857000 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-301: Document Oak
Memoirs in Garbage Collection (WIP):
- Describe usage of MemoryNodeBuilder in Compactor to handle many direct child nodes and track the stableId

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857001 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8111: Create read-only DocumentNodeStore for oak-run recovery dry run - remove unused import

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857010 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8148: [Indexing] Implement sling metric to calculate number of slow queries(relative to all queries)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857057 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8200: MongoDocumentStore in ReadOnly mode should never modify persistence

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857104 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-8202 - Print a warn message when RemoteBlobProcessor swallows an exception

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857109 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8205: Add benchmark for Node.isNodeType()

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857159 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.12.0

release notes



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857165 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8206: Revert Jackrabbit dependency from 2.19.1 to last stable (2.18.0) for stable Oak release

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857178 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
aesthetical fix to release notes heading

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857187 13f79535-47bb-0310-9956-ffa450edef68

mduerig committed:
OAK-301: Document Oak
Memoirs in Garbage Collection (WIP):
- Cleanup before compaction in Oak 1.10

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857188 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.12.0

release notes



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857199 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.12.0

release notes



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857205 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.12.0

release notes



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857206 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.12.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857207 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857209 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8207: Read-only DocumentNodeStore tries to create root document

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857212 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8208: oak-run/rdb: add --rdbtableprefix option

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857221 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8207: Read-only DocumentNodeStore tries to create root document

(fix typo in exception message)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857238 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8141: Replace String path with custom data type

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857240 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8212 : ImporterImpl.importProperties prone to NPE

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857242 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8146: oak-run support for inspecting clusterNodeInfo

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857247 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-7691: Remove deprecated ValueFactoryImpl methods

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857248 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8217: Remove unused field DocumentNodeStoreBuilder#FAST_DIFF

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857252 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8201: RDBDocumentStore in ReadOnly mode should never modify persistence

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857253 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8219: Incorrect parent POM version in oak-doc and oak-doc-railroad-macro

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857254 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8216 : Consistently add nullability annotations with spi.nodetype package and implementation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857255 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8220: CommitRootUpdateTest creates malformed value

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857294 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8215: Use CacheLIRS as default

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857297 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8218: ReadOnlyNodeTypeManager.isNodeType prone to NPE

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857304 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8209: Improve Node.isNodeType(String) performance

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857306 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8214: RDBDocumentStore may not inherit ReadOnly flag from DocumentNodeStore

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857314 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
OAK-8223 - javadoc-plugin fails with java11

- updating the plugin
- removing Adobe's jcr API a weird redirect (MJAVADOC-539) it's building
for me this way.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857332 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8229 : LoginModuleImpl.commit will end in NPE if credentials are null

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857352 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-8235 - upgrade Solr to 6.6.6

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857463 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-8235 - upgrade Solr to 6.6.6

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857480 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8190 : Dedicated authorization for system users (wip)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857551 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.12.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857559 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8231 : Unreachable code in LoginModuleImpl.getLoginId

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857565 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8228: Optimize PathUtils.dropIndexFromName(String)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857569 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8241: Add benchmark for transient Node.setProperty()

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857577 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8232: Node#setPrimaryType(String) does not create child nodes defined as autoCreated

NodeImpl will now auto-create child nodes when a new primary type requires that.


git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857589 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8173 : Add branch coverage to Jacoco check (building auth-external fails on windows)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857591 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8232: Node#setPrimaryType(String) does not create child nodes defined as autoCreated

Removed unused import.


git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857592 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8246: Test failure: NodeSetPrimaryTypeTest.testSetPrimaryType()

Add test to know issues list for now

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857627 13f79535-47bb-0310-9956-ffa450edef68

tteofili committed:
OAK-8235 - upgrade dependencies and imports for oak-solr-osgi

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857634 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8247 Add non-admin mode for Node.isNodeType() benchmark



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857635 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8139: DocumentDiscoveryLiteService hasBacklog silencing must support maven version format

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857638 13f79535-47bb-0310-9956-ffa450edef68

trekawek committed:
OAK-8185: Improve CompositeNodeStore fixture

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857640 13f79535-47bb-0310-9956-ffa450edef68

trekawek committed:
OAK-8185: Improve CompositeNodeStore fixture

- remove unused parameters

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857641 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8197: Update to Jackrabbit 2.18.1

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857687 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8255: update Apache parent pom to version 21

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857723 13f79535-47bb-0310-9956-ffa450edef68

dulceanu committed:
OAK-8243 - Expose the number of SNFEs as metric

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857739 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8256: PrincipalManagerImplTest uses Guava beta methods

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857740 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8258: Active deletion can delete blobs despite indexing cycle deleting them failed

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857936 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-7065: Remove orphan file from local directory in case indexing fails

Test to assert that local directory created due to reindexing get
cleaned up if indexing cycle fails

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857937 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-301: Document Oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857986 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-301: Document Oak

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857989 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-7065: Remove orphan file from local directory in case indexing fails

Fixing oak-run build failure (OAK-8259) due to changes done in r1857937.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857993 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-7256 When query with OR is divided into union of queries, options (like index tag) are not passed into subqueries

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857996 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8240 : Improve test coverage for org.apache.jackrabbit.oak.security.authorization.permission

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1857999 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8265 : Add nullable/notnull annotations to org.apache.jackrabbit.oak.security.authorization.permission.*

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858005 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8184 With statistical mode, facet count seems having higher error rate than expected

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858008 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8258: Active deletion can delete blobs despite indexing cycle deleting them failed

Setup indexing failed flag after merge completes successfully instead of
catching a class of exception - this should be more resilient in other
kind of failures while indexing.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858032 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8270 : Add variant of PermissionUtil.getReadOnlyTree that never returns null

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858042 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8272: Orphaned branch commit entries after restart

Add ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858047 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8272: Orphaned branch commit entries after restart

Commit fix and enable test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858052 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8257: RDBDocumentStore: improve trace logging of batch operations

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858053 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8234 Reduce object allocation in PermissionProviderImpl for admin sessions



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858054 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8234 : adding missing annotations with AllPermissionProviderImpl

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858061 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8276 : OR-CompositeTreePermission.grantsPermission should loop over aggregates of supported permissions

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858099 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8277: Run tests on travis-ci.org on MongoDB 4.0.x

Update to MongoDB 4.0.9

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858123 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8277: Run tests on travis-ci.org on MongoDB 4.0.x

Update documentation about recommended MongoDB version

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858124 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8273: RDBDocumentStore: createOrUpdate with less than 3 ops suboptimal

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858139 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8281 : Improve coverage for org.apache.jackrabbit.oak.security.authorization.composite

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858190 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8190: Javadoc fixes

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858194 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8278: RDBDataSourceFactory.CloseableDataSource doesn't implement java.sql.Wrapper correctly

CloseableDataSource now wraps it's internal DataSource instance.


git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858195 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8263 : Simplify PermissionEntry.compareTo

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858196 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8278: RDBDataSourceFactory.CloseableDataSource doesn't implement java.sql.Wrapper correctly

Fixed implementation to respect Wrapper contract.


git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858197 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8266 : Redundant check for existing tree in CompiledPermissionImpl.buildVersionDelegatee

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858199 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8286: Update jetbrains nullability annotations to 17.0.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858385 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8149: java 12 javadoc fails with 'unknown tag' for OSGi version annotations in package-info - added workaround to javadoc plugin config

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858407 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8269 : Improve readability of CompositePermissionProvider and CompositeTreePermission

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858410 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8284 : Improvements to EntryPredicate

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858411 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8288: fix javadoc:javadoc for jdk >= 13

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858414 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8288: fix javadoc:javadoc for jdk >= 13

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858415 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8288: fix javadoc:javadoc for jdk >= 13

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858416 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8288: fix javadoc:javadoc for jdk >= 13

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858417 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8288: fix javadoc:javadoc for jdk >= 13

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858418 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8290: Update org.apache.felix.framework for jdk13

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858424 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8288: fix javadoc:javadoc for jdk >= 13 - changes reverted (see ticket)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858442 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8249 NodeImpl#isNodeType could load mixin info lazily



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858530 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8293 : PrincipalProviderImpl.findPrincipals my return null values in result iterator

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858534 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8262 : CompiledPermissionImpl returns wrong type of TreePermission for child of permission store 

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858538 13f79535-47bb-0310-9956-ffa450edef68

catholicon committed:
OAK-8166: Index definition with orderable property definitions with and without functions breaks index

Committing fix contributed by Nitin Gupta (@nitigup).

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858544 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8296: DocumentNodeStoreBranchesTest uses javax.annotation.Nonnull

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858571 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-7902: Update osgi-mock to 2.3.10

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858578 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8283 Make LazyValue implement Supplier



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858758 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8240: Improve test coverage for org.apache.jackrabbit.oak.security.authorization.permission

Remove one bogus and several unused imports

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858774 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8300: Revision GC may remove previous document without removing reference

Add ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858808 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8301: Ensure travis-ci uses trusty image

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858810 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8187 : Respect service ranking with AuthorizableNodeName, AuthorizableActionProvider and RestrictionProvider

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858830 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8300: Revision GC may remove previous document without removing reference

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858838 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8269 : Improve readability of CompositePermissionProvider and CompositeTreePermission (redundant code in CompositePermissionProviderOr.getPrivileges)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858846 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8305 Don't log a warning for queries that use option(traversal fail)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858913 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8307: RDBDocumentStore: add DEBUG logging when fetching index metadata fails

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858926 13f79535-47bb-0310-9956-ffa450edef68

stillalex committed:
OAK-8306 Empty PrincipalProvider cache breaks membership collection



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858931 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8304 : VersionablePathHook: Diff.setVersionablePath never results in CommitFailedException

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1858994 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8310: Potentially misleading conflict exception message

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859020 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8240 : Improve test coverage for org.apache.jackrabbit.oak.security.authorization.permission

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859062 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8308 : Unused Object methods in PermissionStoreEditor.AcEntry

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859063 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8295 : Update version of jackrabbit dependency

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859065 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8190 : Dedicated authorization for system users (replace PrincipalPolicy by Jackrabbit API)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859071 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8311: RDBDocumentStore: allow to turn off RDB-specific MissingLastRevSeeker

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859231 13f79535-47bb-0310-9956-ffa450edef68

trekawek committed:
OAK-8317: Move the version-related migration tools to the oak-core

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859273 13f79535-47bb-0310-9956-ffa450edef68

trekawek committed:
OAK-8318: Allow to close the SegmentStoreMigrator

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859274 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8315 : Missing nullability annotations in package o.a.j.oak.security.authentication.token and o.a.j.oak.spi.security.authentication.token

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859279 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8312: MissingLastRevSeeker: improve progress logging

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859292 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8312: NodeDocumentSweeper: improve progress logging

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859294 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
OAK-8314: BlobIds created with direct uploads lack length suffix

Suffixing length to the ids being returned for for directly uploded binaries

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859350 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
OAK-8314: BlobIds created with direct uploads lack length suffix

Enable DocumentNodeStore-Mongo fixture for BinaryAccessDSGCIT

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859351 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8320 : Improve tests for o.a.j.oak.security.authentication.token package

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859352 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8321 : Use format specifier for log output

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859353 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8246: Test failure: NodeSetPrimaryTypeTest.testSetPrimaryType() - fixed by updated Jackrabbit test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859359 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8316 : Drop userId field in TokenLoginModule

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859360 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
OAK-8314: BlobIds created with direct uploads lack length suffix

Proper DB cleanup for DocumentNodeStore-Mongo fixture for BinaryAccessDSGCIT

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859365 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8324 : Missing NotNull annotation with AuthenticationConfiguration and LoginModuleStatsCollector 

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859377 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8322: Add oakathon.md documentation page


git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859423 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8294 Make it possible to mark a (customer) query as bad and skip it

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859424 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8322: add license

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859426 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859427 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8323 : Refactor redundant checks for null in TokenValidator

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859429 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8294 Make it possible to mark a (customer) query as bad and skip it

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859430 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8309 SLOW_QUERY_COUNT don't get updated for each slow query

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859436 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8329: update docker-junit-rule dependency

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859437 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8328 | Config errors in index defintion causes index update to get stuck in a loop

reschke committed:
OAK-8331: Update Tika dependency to 1.21

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859533 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-301: Document Oak

Update wiki links to point to the archived pages
Link to the JCR 2.0 specification hosted at adobe.com

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859561 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller commented on oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUpdate.java:
```diff
@@ -276,9 +276,19 @@ private void collectIndexEditors(NodeBuilder definitions,
+                    // where multiple values are assigned to a property that is supposed to be single valued
+                    // We log an error message here and continue - this way the bad index defintion is ignored and doesn't block the async index update
+
+                    log.error("Unable to get Index Editor for index at {} . Please correct the index definition " +
```
I would also log the stack trace of the root cause, that is in this case append ", e" to the parameters: log.error(..., e)

thomasmueller commented on oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/util/ConfigUtil.java:
```diff
@@ -36,34 +36,71 @@
  */
 public class ConfigUtil {
 
+    private static final String ILLEGAL_STATE_EXCEPTION_ERROR_MESSAGE = "Multiple Values provided for property %s in Index Defintion . Single value was expected";
```
There are some typos, and some words are uppercase. Could you fix it please.

thomasmueller commented on oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/util/ConfigUtil.java:
```diff
@@ -36,34 +36,71 @@
+        try{
+            PropertyState ps = definition.getProperty(propName);
+            return ps == null ? defaultVal : ps.getValue(Type.BOOLEAN);
+        }catch(IllegalStateException e ){
```
Please add spaces after "}", before "{", around "()", and remove the superfluous space after e.

thomasmueller replied:
What I usually do is use the IDE formatter, but only for the selected text. (Eclipse allows this, and I'm almost sure IntelliJ as well.)

reschke committed:
OAK-8332: update Tomcat JDBC dependency to 8.5.41

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859609 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller commented on oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUpdate.java:
```diff
@@ -276,9 +276,19 @@ private void collectIndexEditors(NodeBuilder definitions,
+                try {
+                    editor = rootState.provider.getIndexEditor(type, definition, rootState.root,
+                            rootState.newCallback(indexPath, shouldReindex, getEstimatedCount(definition)));
+                } catch (IllegalStateException e) {
```
Would it be possible to log the problem where the IllegalStateException is thrown, that is in the IndexDefinition.java file? Maybe in the builder... I just think that the IndexUpdate class is more far away than necessary from there the exception is thrown. I think it is possible to move it closer to index definition building. Still it might re-throw an exception. It's just that in the future, there might be other reasons why IllegalStateException that are not related to a wrong index definition itself (maybe for a remote index).

nit0906 replied:
@thomasmueller  - Sure - I will log the problem in IndexDefinition . I will rethrow the exception from there - and in the catch in the IndexUpdate  and just skip the iteration for this index (without logging the error) Or maybe log error at both the places - IndexUpdate as well as IndexDefintion

reschke committed:
OAK-8334: Update Jackson dependency to 2.9.9

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859612 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Merge branch 'upstream-trunk' into OAK-8328

reschke committed:
OAK-8335: get rid of junit deprecation warnings

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859619 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8337: RDBDocumentStore: refactor index dumping code

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859711 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8328 Incorporating review comments

reschke committed:
OAK-8338: RDBDocumentStoreJDBC: fix theoretically possible NPE in perflogging code

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859716 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8341: Include tomcat-jdbc/juli in oak-run

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859772 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8004: oak-run: support recovery command for RDBDocumentStore - update documentation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859776 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8111: Create read-only DocumentNodeStore for oak-run recovery dry run - update documentation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859780 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8340: Document use of oak-run for RDB

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859792 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8340: Document use of oak-run for RDB - fix typo

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859794 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8146: oak-run support for inspecting clusterNodeInfo - add documentation and fix usage string for --clusterId

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859843 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8345: oak-run: cleanup command list and documentation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859854 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8346: RDBDocumentStore*: fix several potential but improbable NPEs

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1859881 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8348: Update surefire/failsafe dependencies to 2.22.2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860120 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8350: Update animal-sniffer dependency to 1.18

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860131 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8349: RDBDocumentStore*: "reset clusterId tool" in oak-run - whitespace cleanup

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860137 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8261 Indexing lane failing but the index is not marked corrupt (formatting changes)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860192 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8227 : Performance benchmarks (wip)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860199 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8261 Indexing lane failing but the index is not marked corrupt

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860200 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8353: Additional test for OAK-8012

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860202 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8321 : Use format specifier for log output (in o.a.j.o.security.user)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860216 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8354 : Fix/Add nullability annotations with o.a.j.oak.security.user package

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860234 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8248: Update Oak trunk and 1.10 to Jackrabbit 2.18.2

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860235 13f79535-47bb-0310-9956-ffa450edef68

trekawek committed:
OAK-8357: Support the truncate operation for the GCJournalFile

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860236 13f79535-47bb-0310-9956-ffa450edef68

trekawek committed:
OAK-8356: Support append mode in the SegmentStoreMigrator

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860237 13f79535-47bb-0310-9956-ffa450edef68

dulceanu committed:
OAK-8358 - oak-run check should have an option for specifying memory mapping

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860247 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
fix svn:eol-style

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860250 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8354 : Fix/Add nullability annotations with o.a.j.oak.security.user package

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860253 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8355 : AuthorizableImpl.isEveryone should not throw RepositoryException

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860254 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8360 : UserAuthentication.authenticate: improve readability

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860270 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-8363: Remove obsolete comment

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860278 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8362 : UserImporter: redundant check for null principals

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860280 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8365 UserManagerImpl: unreacheable code when retrieving user/group from ID and tree

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860319 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8368: RDBDocumentNodeStoreBuilder: refactor setRDBConnection for consistency

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860328 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8369:Misplace checkNotNull in CacheValidator

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860329 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8251: RDBBlobStore instantiated from ReadOnly DocumentNodeStore should never modify persistence - add ignored test

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860330 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8370 : Simplify o.a.j.oak.security.user.Utils.getOrAddTree

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860338 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8370 : Simplify o.a.j.oak.security.user.Utils.getOrAddTree (and fix javadoc)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860339 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8372 : DeclaredMembershipPredicate contains redundant lookup of groupTree

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860349 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8373 : UserPrincipalProvider: redundant test for null in ResultRowToPrincipal, EveryonePredicate

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860351 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8361 : Improve tests for o.a.j.oak.security.user package

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860352 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8371: Stop using deprecated DocumentMK.Builder in RDB tests

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860355 13f79535-47bb-0310-9956-ffa450edef68

mreutegg committed:
OAK-301: Document Oak

Clarify content model

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860356 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8375: DocumentNodeStoreBuilder should not instantiate BlobStore instance before "build()"

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860393 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8251: RDBBlobStore instantiated from ReadOnly DocumentNodeStore should never modify persistence

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860442 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8376: update commons-codec dependency to 1.12

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860548 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
minor improvement: unused imports

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860549 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8321 : Use format specifier for log output

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860550 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8377 : Jacoco plugin: add PACKAGE based rules for security code

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860562 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8349: RDBDocumentStore*: "reset clusterId tool" in oak-run

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860564 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8378: rdb/oak-run: update usage and documentation for garbage command

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860565 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8379 : Improve tests for o.a.j.oak.security.authorization

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860566 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.14.0

release notes



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860625 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8383 : AccessControlValidator: check for duplicate ACE ignores allow/deny status

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860653 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8386 : AccessControlValidator.checkValidAccessControlEntry: redundant test for tree existence 

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860656 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8255: update Apache parent pom to version 21 - temporarily reverted

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860660 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit Oak 1.14.0

release notes



git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860666 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.14.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860667 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
[maven-release-plugin] prepare for next development iteration

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860669 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8385 : Get rid of duplicated AcEntry/Entry in PermissionStoreEditor and AccessControlValidator

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860703 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8388 : AccessControlManagerImpl.getEffectivePolicies(Set): Insufficient validation of query results

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860715 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8389 : AccessControlValidator prone to NPE

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860716 13f79535-47bb-0310-9956-ffa450edef68

thomasmueller committed:
OAK-8344 Expose local index directory size as a metric

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860717 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8390 : ACL.java : improve readability

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860720 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8155 : CompositePermissionProvider: add possibility to abort evaluation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860721 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8392 : AcListComparator only compares path

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860729 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8384 : Improve tests for o.a.j.oak.security.authorization.accesscontrol
minor improvement: fix indention with AccessControlValidator

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860730 13f79535-47bb-0310-9956-ffa450edef68

amit-jain committed:
OAK-8394: Fix BinaryAccessDSGCIT failing intermittently for Mongo

Catching exception if any to ignore tests and also moved init of mongo connection factory only when datastore available.

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1860931 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-8366 - Add monitoring for Azure Storage

Contribution by Ieran Draghiciu

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861013 13f79535-47bb-0310-9956-ffa450edef68

francescomari committed:
OAK-8366 - Add license headers

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861016 13f79535-47bb-0310-9956-ffa450edef68

davidegiannella committed:
Apache Jackrabbit oak 1.14.0

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861027 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Merge branch 'upstream-trunk' into OAK-8328

reschke committed:
OAK-8396: documentation steps missing to run sweep on RDBDocumentStore

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861114 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8391 : Create AggregationFilter implementation

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861116 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8227 : Performance benchmarks (respecting aggregationfilter option)

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861117 13f79535-47bb-0310-9956-ffa450edef68

anchela committed:
OAK-8397 : maven-bundle-plugin: add export instructions with baseline 1.14

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861127 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8339: Move jackrabbit-api project into Oak - copy source and minimally adjust the POM

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861128 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8339: Move jackrabbit-api project into Oak - adjust README

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861129 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8339: Move jackrabbit-api project into Oak - add to reactor

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861150 13f79535-47bb-0310-9956-ffa450edef68

Commit:
OAK-8328 | Adding test and fixing a failing test

anchela committed:
OAK-8401 : Expand security topic in side bar

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861233 13f79535-47bb-0310-9956-ffa450edef68

reschke committed:
OAK-8339: Move jackrabbit-api project into Oak - change dependencies

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/oak/trunk@1861235 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Merge branch 'upstream-trunk' into OAK-8328

Commit:
OAK-8328 | trivial change
